consul/consul.go

```diff
@@ -110,7 +116,8 @@ func (c *Consul) getAllServices(agent *consulAPI.Client) ([]*service.Service, er
 	for _, dcAwareQuery := range dcAwareQueries {
 		consulServices, _, err := agent.Catalog().Services(dcAwareQuery)
 		if err != nil {
-			return nil, err
+			log.WithError(err).Error("An error occurred getting services from Consul, will continue with another DC")
```

Comment:
Does it actually check another DC if there is only one DC query (when a DC is specified in `dcAwareQueriesForAllDCs`)? Same goes for line 251. BUT just change the text if checking this involves too much overhead.

Reply:
there is always for loop over all DCs that Consul knows about (variable dcAwareQueries) https://github.com/allegro/marathon-consul/blob/consuldatacenters/consul/consul.go#L91 since Marathon in most of the deployements is not stretched across multiple DCs this logic was a bit wrong from the start ;)

Reply:
Do we add a metric for that?